eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/model/SubSnpNoHgvs.java

```diff
@@ -195,4 +245,123 @@ public boolean isAssemblyMatch() {
+        String[] dividedAlleles = StringUtils.split(alleles, "/");
+
+        if (allelesOrientation.equals(Orientation.FORWARD)) {
+            return dividedAlleles;
+        } else if (allelesOrientation.equals(Orientation.REVERSE)) {
+            return getReversedComplementedAlleles(dividedAlleles);
+        } else {
+            throw new IllegalArgumentException(
+                    "Unknown alleles orientation " + allelesOrientation + " for variant " + this);
+        }
+    }
+
+    private Orientation getAllelesOrientation() {
+        return Orientation.getOrientation(
+                this.subsnpOrientation.getValue() * this.snpOrientation.getValue() * this.contigOrientation.getValue());
+    }
+
+    private String[] getReversedComplementedAlleles(String[] alleles) {
+        for (int i=0; i < alleles.length; i++) {
+            alleles[i] = calculateReverseComplement(alleles[i]);
+        }
+        return alleles;
+    }
+
+    private static String calculateReverseComplement(String alleleInReverseStrand) {
+        StringBuilder alleleInForwardStrand = new StringBuilder(alleleInReverseStrand).reverse();
+        for (int i = 0; i < alleleInForwardStrand.length(); i++) {
+            switch (alleleInForwardStrand.charAt(i)) {
+                // Capitalization holds a special meaning for dbSNP so we need to preserve it.
+                // See https://www.ncbi.nlm.nih.gov/books/NBK44414/#_Reports_Lowercase_Small_Sequence_Letteri_
+                case 'A':
+                    alleleInForwardStrand.setCharAt(i, 'T');
+                    break;
+                case 'a':
+                    alleleInForwardStrand.setCharAt(i, 't');
+                    break;
+                case 'C':
+                    alleleInForwardStrand.setCharAt(i, 'G');
+                    break;
+                case 'c':
+                    alleleInForwardStrand.setCharAt(i, 'g');
+                    break;
+                case 'G':
+                    alleleInForwardStrand.setCharAt(i, 'C');
+                    break;
+                case 'g':
+                    alleleInForwardStrand.setCharAt(i, 'c');
+                    break;
+                case 'T':
+                    alleleInForwardStrand.setCharAt(i, 'A');
+                    break;
+                case 't':
+                    alleleInForwardStrand.setCharAt(i, 'a');
+                    break;
+            }
+        }
+        return alleleInForwardStrand.toString();
+    }
+
+    private String getMicrosatelliteAlternate(String alternate, String firstAltAllele) {
+        if (NumberUtils.isDigits(alternate)) {
+            Matcher matcher = microsatellitePattern.matcher(firstAltAllele);
+            if (matcher.matches()) {
+                alternate = matcher.group(STR_SEQUENCE_REGEX_GROUP) + alternate;
+            } else {
+                throw new IllegalArgumentException("Not parseable STR: " + reference + "/" + alternate);
+            }
+        }
```

Comment:
Agreed offline with @pabarcgar that this might need a separate pull request considering some of the special cases laid out [here](https://www.ebi.ac.uk/panda/jira/browse/EVA-1220?focusedCommentId=246718&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-246718)